lstchain/onsite.py

```diff
@@ -171,3 +188,99 @@ def find_systematics_correction_file(pro, date, sys_date=None, base_dir=DEFAULT_
     selected_date = next((day for day in sys_date_list if day <= date), sys_date_list[-1])
 
     return (sys_dir / selected_date / pro / f"ffactor_systematics_{selected_date}.h5").resolve()
+
+def find_calibration_file(pro, calibration_run=None, date=None, category=DataCategory.A, base_dir=DEFAULT_BASE_PATH):
+
+    if category == DataCategory.A:
+        cal_dir = Path(base_dir) / CAT_A_PIXEL_DIR / "calibration"
+    elif category == DataCategory.C:
+        cal_dir = Path(base_dir) / CAT_B_PIXEL_DIR / "calibration"
+    else:
+        raise ValueError(f'Argument \'category\' can be only \'DataCategory.A\' or \'DataCategory.B\', not {category}')    
+
+    if calibration_run is None and date is None:
+        raise ValueError('Must give at least `date` or `run`')
+
+    if calibration_run is not None:
+        # search a specific calibration run
+        file_list = sorted(cal_dir.rglob(f'{pro}/calibration*.Run{calibration_run:05d}.0000.h5'))
+
+        if len(file_list) == 0:
+            raise IOError(f"Calibration file from run {calibration_run} not found\n")
+
+        return file_list[0].resolve()
+
+    # search for a unique calibration file from the same date
+    file_list = sorted((cal_dir / date / pro).glob('calibration*.0000.h5'))
+    if len(file_list) == 0:
+        raise IOError(f"No calibration file found for date {date}")
+
+    if len(file_list) > 1:
+        raise IOError(f"Too many calibration files found for date {date}: {file_list}, choose one run\n")
+
+    return file_list[0].resolve()
+
+def find_DL1_subrun(run, sub_run, dl1_dir=DEFAULT_DL1_PATH):
+    '''
+    Find the given subrun DL1 file (i.e. globbing for the date part)
+    '''
+    file_list = rglob_symlinks(dl1_dir, f'dl1_LST-1.1.Run{run:05d}.{sub_run:04d}*.h5')
+    # ignore directories that are not a date, e.g. "Trash"
+    file_list = [p for p in file_list if is_date(p.parent.name)]
+
+    if len(file_list) == 0:
+        raise IOError(f"Run {run} not found\n")
+
+    if len(file_list) > 1:
+        raise IOError(f"Found more than one file for run {run}.{sub_run}: {file_list}")
+
+    return file_list[0]
+
+def find_interleaved_subruns(run, dl1_dir=DEFAULT_DL1_PATH):
+    '''
+    Find the given subrun interleaved file (i.e. globbing for the date part)
+    '''
+    file_list = rglob_symlinks(dl1_dir, f'interleaved_LST-1.Run{run:05d}.*.h5')
```

Comment:
Are the interleaved files supposed to be stored in the same place as the DL1 subrun-wise ones? We could store them in a different subdirectory for clarity

Reply:
Yes, for moment, they are together with the muons, we could indeed move both in a specific directory, e.g. 
/fefs/aswg/data/real/DL1/[date]/[version]/interleaved
/fefs/aswg/data/real/DL1/[date]/[version]/muons